src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedDefaultTests.cs

```diff
@@ -292,7 +293,33 @@ static bool M<T>(T x = default)
             Assert.Equal(Conversion.Identity, model.GetConversion(default2));
         }
 
+        [Fact, WorkItem(40791, "https://github.com/dotnet/roslyn/issues/40791")]
+        public void ComparisonWithGenericType_Unconstrained_Inequality()
+        {
+            string source = @"
+class C
+{
+    static bool M<T>(T x = default)
```

Comment:
>T [](start = 18, length = 1)

Would presence of a class constraint make a difference? Do we have  a test?

Reply:
It does. See `ComparisonWithGenericType_ReferenceType`. `TRef = default` is okay.
I'll add a test for more permutations

---
In reply to: [364496957](https://github.com/dotnet/roslyn/pull/40801#discussion_r364496957) [](ancestors = 364496957)